Macros.h

```diff
@@ -27,6 +26,14 @@
 #define SPACESHIP_NULLSTR(a, b)   strcmp((a) ? (a) : "", (b) ? (b) : "")
 #endif
 
+#ifndef HTOP_ASSERT
+#ifndef NDEBUG
+#define HTOP_ASSERT(expr)         ((expr) ? ((void)0) : (CRT_done(), assert(expr)))
```

Comment:
This may re-evaluate `expr` twice …

According to documentation `assert(3)` calls `abort(3)` which `raise(3)`s some `SIGABRT`. Maybe catch things from that end?

Reply:
One should not pass an expression to `assert(3)`, cause it might (depending on `NDEBUG` being defined or not) evaluated once or not at all.
So currently the expression gets evaluated 0 time in normal mode and 1 time in debug mode.
With the new wrapper it would get evaluated 0 time in normal mode, 1 time in debug mode if true and 2 times in debug mode if false.
The main point of using `assert(3)` instead of `raise(SIGABRT)` is the message formatting of `assert(3)` (which we otherwise would need to replicate).